metrics-dropwizard/src/main/java/com/mesosphere/usi/metrics/dropwizard/GarbageCollectorMetricSet.java

```diff
@@ -0,0 +1,52 @@
+package com.mesosphere.usi.metrics.dropwizard;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricSet;
+
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import java.util.*;
+import java.util.regex.Pattern;
+
+import static com.codahale.metrics.MetricRegistry.name;
+
+/**
+ * A set of gauges for the counts and elapsed times of garbage collections.
+ *
+ * This is a copy of GarbageCollectorMetricSet from Dropwizard metrics, slightly adjusted to support USI metric name
+ * conventions.
+ */
+public class GarbageCollectorMetricSet implements MetricSet {
+    private static final Pattern WHITESPACE = Pattern.compile("[\\s]+");
+
+    private final List<GarbageCollectorMXBean> garbageCollectors;
+
+    /**
+     * Creates a new set of gauges for all discoverable garbage collectors.
+     */
+    public GarbageCollectorMetricSet() {
+        this(ManagementFactory.getGarbageCollectorMXBeans());
+    }
+
+    /**
+     * Creates a new set of gauges for the given collection of garbage collectors.
+     *
+     * @param garbageCollectors the garbage collectors
+     */
+    public GarbageCollectorMetricSet(Collection<GarbageCollectorMXBean> garbageCollectors) {
+        this.garbageCollectors = new ArrayList<>(garbageCollectors);
+    }
+
+    @Override
+    public Map<String, Metric> getMetrics() {
+        final Map<String, Metric> gauges = new HashMap<>();
+        for (final GarbageCollectorMXBean gc : garbageCollectors) {
+            final String name = WHITESPACE.matcher(gc.getName()).replaceAll("-").toLowerCase(Locale.US);
+            gauges.put(name(name, "collections.gauge"), (Gauge<Long>) gc::getCollectionCount);
+            gauges.put(name(name, "collections.duration.gauge.seconds"),
+                    (Gauge<Double>) () -> (new Long(gc.getCollectionTime()).doubleValue()) / 1000.0);
```

Comment:
this is a crazy line of code... 
1. get double value from `gc.getCollectionTime()`
2. convert that double to a `Long`
3. get the `doubleValue` of that long value that was a double
4. divide by a float to be autoboxed into a `Double` part of `Gauge<Double>`